### ISSUE TITLE
[CM] Management merge Reloadable and ReloadableList into a single Reloadable

### ISSUE THREAD

**Issue by ph** [elastic/beats#10021]
In CM we have two concepts of reloadable, a **single** like the output, or a list like the **modules**, for me these are implementations details and we should uniform them into a single interface `Reloader` that takes reload request and do the right thing. It is also important to keep trace or have hook to when a request fails so we can report the error back to the API.

Maybe we should have something in between the current Manager and the Reloader that can take the current state of the system (a snapshot) and create the necessary steps to migrate to the next new state of the system.  

This mean that Autodiscover and the current manager (name would need to change) are only creating requests to a coordinator responsable in keeping the correct state.

Ref: https://github.com/elastic/beats/pull/10019

**Comment by jalvz** [comment]
There is different logic for objects (last write wins) and lists (merge). If they are unified how will that work?

**Comment by ph** [comment]
I think the main problem is the feedback of what _was applied_ what is _not applied_. Merging both of concept appears to be difficult at first, it's because we do have multiple ways to starts parts of beats.

Before merging both of the concepts we need to merge where the reload and the managing of instances happens (input + output + etc).

I see that YAML configuration, modules configuration and CM configuration should just submit requests to a single entry point, I see it as a transaction it could be atomic or not depending at how we decide to implement it.

So we could do the following:

- Submit change request
- System derive what need to be done before converging.
- System can validate if actions are problematics (double ouput?)
- System applies the changes.
- System reports the status of the changes.

**Comment by ph** [comment]
I am going to close this as wontfix for now, we are looking at changing the architecture and this doesn't fit.
